beacon-chain/db/kv/state.go

```diff
@@ -196,6 +204,14 @@ func (k *Store) DeleteState(ctx context.Context, blockRoot [32]byte) error {
 			}
 		}
 
+		slot, err := slotByBlockRoot(ctx, tx, blockRoot[:])
```

Comment:
So if there is no block by that root, for whatever reason, we can never delete this state?